Sources/Console/Terminal/Terminal.swift

```diff
@@ -58,60 +72,61 @@ public class Terminal: ConsoleProtocol {
         return readLine(strippingNewline: true) ?? ""
     }
 
-    public func execute(_ command: String) throws {
-        #if os(Linux)
-            let input = FileHandle.standardInput()
-            let output = FileHandle.standardOutput()
-            let error = FileHandle.standardError()
+    public func execute(program: String, arguments: [String], input: Int32? = nil, output: Int32? = nil, error: Int32? = nil) throws {
+        var pid = UnsafeMutablePointer<pid_t>.allocate(capacity: 1)
+        pid.initialize(to: pid_t())
+
+        let args = [program] + arguments
+        let argv: [UnsafeMutablePointer<CChar>?] = args.map{ $0.withCString(strdup) }
+        defer { for case let arg? in argv { free(arg) } }
+
+        var environment: [String: String] = [:]
+        #if Xcode
+            let keys = ["SWIFT_EXEC", "HOME", "PATH", "TOOLCHAINS", "DEVELOPER_DIR", "LLVM_PROFILE_FILE"]
         #else
-            let input = FileHandle.standardInput
-            let output = FileHandle.standardOutput
-            let error = FileHandle.standardError
+            let keys = ["SWIFT_EXEC", "HOME", "PATH", "SDKROOT", "TOOLCHAINS", "DEVELOPER_DIR", "LLVM_PROFILE_FILE"]
         #endif
-        try execute(command, input: input, output: output, error: error)
-    }
-
-    public func subexecute(_ command: String, input: String) throws -> String {
-        let output = Pipe()
-        let input = Pipe()
-        let error = Pipe()
 
-        do {
-            try execute(command, input: input, output: output, error: error)
-        } catch ConsoleError.execute(let result) {
-            let error = String(data: error.fileHandleForReading.readDataToEndOfFile(), encoding: .utf8) ?? "Unknown"
-            throw ConsoleError.subexecute(result, error)
+        func getenv(_ key: String) -> String? {
+            let out = libc.getenv(key)
+            return out == nil ? nil : String(validatingUTF8: out!)  //FIXME locale may not be UTF8
```

Comment:
You could also do `out.flatMap { String(validatingUTF8: $0) }` and remove the `!`